### PR TITLE
16175 Fixed undefined legal name in a dissolution (and also restoration)

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#16175

*Description of changes:*
- app version = 5.2.1
- deleted setEntityName and used setLegalName instead
- deleted getBusiness and used getBusinessOfficeAddress instead
- fixed undefined legal name -- added fallback for parsing draft without legal name (or legal type or founding date)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Dissolution/AssociationDetails.vue [hunk 1]
```diff
@@ -27,8 +27,8 @@
         <v-col cols="12" sm="4" class="pr-4 pt-4 pt-sm-0">
           <label class="mailing-address-header">Mailing Address</label>
           <MailingAddress
-            v-if="!isEmptyAddress(getBusiness.officeAddress.mailingAddress)"
```
Comment (severinbeauvais) [inline src/components/Dissolution/AssociationDetails.vue:30]: `getBusiness` was only ever used to get tge office address, so I replaced the general getter with a more specific one.

File: src/mixins/filing-template-mixin.ts [hunk 2]
```diff
@@ -571,9 +572,9 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
     this.setFilingId(+draftFiling.header.filingId)
 
     // restore Business data
-    this.setEntityType(draftFiling.business.legalType)
-    this.setLegalName(draftFiling.business.legalName)
-    this.setFoundingDate(draftFiling.business.foundingDate)
+    this.setEntityType(draftFiling.business.legalType || this.getBusinessLegalType)
+    this.setLegalName(draftFiling.business.legalName || this.getBusinessLegalName)
+    this.setFoundingDate(draftFiling.business.foundingDate || this.getBusinessFoundingDate)
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:577]: Because the draft restoration (or dissolution, below) didn't have legal name, it was being cleared out here. (It was actually loaded via a business fetch in App.vue, so this code here is probably not needed, but that cleanup is out of scope for this quick fix.)

**Review by JazzarKarim — APPROVED** [review]
LGTM!
